client/lib/wp/sync-handler/index.js

```diff
@@ -20,6 +20,46 @@ const localforage = getLocalForage();
 const debug = debugFactory( 'calypso:sync-handler' );
 
 /**
+ * Generate a key from the given param object
+ *
+ * @param {Object} params - request parameters
+ * @param {Boolean} applyHash - codificate key when it's true
+ * @return {String} request key
+ */
+export const generateKey = ( params, applyHash = true ) => {
+	var key = `${params.apiVersion || ''}-${params.method}-${params.path}`;
+
+	if ( params.query ) {
+		key += '-' + params.query;
+	}
+
+	if ( applyHash ) {
+		key = new Hashes.SHA1().hex( key );
+	}
+
+	key = 'sync-record-' + key;
+
+	debug( 'key: %o', key );
+	return key;
+}
+
+/**
+ * Detect pagination changes in local vs request response bodies
+ * @param  {Object} res - response object as passed from promise
+ * @param  {Object} localResponseBody - local response body
+ * @return {Boolean} returns whether pagination has changed
+ */
+export const hasPaginationChanged = ( res, localResponseBody ) => {
+	if ( ! res || ! res.meta || ! res.meta.page_handle ) {
+		return false;
+	}
+	if ( localResponseBody && localResponseBody.meta && localResponseBody.meta.page_handle === res.meta.page_handle ) {
```

Comment:
It isn't an object, it's a string. Although I have the attribute wrong... it should be `next_page` rather than `page_handle`. I'll fix that. It's implemented in a couple of places for now, including post-lists, page-lists, and media-lists. Maybe others.